shared/settings/email/container.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import React, {Component} from 'react'
+import UpdateEmail from './index'
+import {connect} from 'react-redux'
+import {navigateUp} from '../../actions/router'
+import {onSubmitNewEmail, onChangeNewEmail} from '../../actions/settings'
+
+import type {Props} from './index'
+import type {TypedState} from '../../constants/reducer'
+
+class UserEmailContainer extends Component<void, Props, void> {
+  static parseRoute () {
+    return {
+      componentAtTop: {title: 'Change Email'},
+    }
+  }
+
+  render () {
+    return <UpdateEmail {...this.props} />
+  }
+}
+
+export default connect(
+  (state: TypedState, ownProps: {}) => {
+    const {emails, errorMessage, newEmail} = state.settings.email
+    if (emails.length > 0) {
+      const email = emails[0].email
+      return {
+        email,
+        isVerified: emails[0].isVerified,
+        edited: newEmail && newEmail !== email,
+        errorMessage,
+      }
+    }
+    return {}
```

Comment:
I'm sorry this connect isn't type checking properly. I need to fix that.
But you need to return the right keys with null values.
Instead of emptyObj, should have `{errorMessage: null, edited: false...}` Needs to match the type def.